### ISSUE TITLE
Switch similarity search to different library

### ISSUE THREAD

**Issue by deepfates** [deepfates/memery#30]
Annoy takes a long time to build, maybe there's something else out there that doesn't

**Comment by deepfates** [comment]
This looks quite useful, thank you! It can do CRUD ops on individual
elements in the index which is desirable.

It does look like it needs to know the maximum possible number of elements
before creating the index, but maybe we can work within that constraint..

**Comment by igorbrigadir** [comment]
Yeah the way I used it last time was just setting an order of magnitude of expected entries, and if you fill up, you can always reindex (which isn't as bad as it sounds, usually)
